--- a/chrome/browser/ui/views/location_bar/selected_keyword_view.cc
+++ b/chrome/browser/ui/views/location_bar/selected_keyword_view.cc
@@ -178,7 +178,7 @@ const std::u16string& SelectedKeywordVie
 
 int SelectedKeywordView::GetExtraInternalSpacing() const {
   // Align the label text with the suggestion text.
-  return 14;
+  return 3.5f;
 }
 
 void SelectedKeywordView::SetLabelForCurrentWidth() {
