--- a/chrome/browser/ui/views/location_bar/selected_keyword_view.cc
+++ b/chrome/browser/ui/views/location_bar/selected_keyword_view.cc
@@ -44,15 +44,7 @@ SelectedKeywordView::GetKeywordLabelName
     bool is_gemini_keyword = false;
     names.short_name = service->GetKeywordShortName(
         keyword, &is_extension_keyword, &is_gemini_keyword);
-    if (is_gemini_keyword) {
-      names.full_name = l10n_util::GetStringFUTF16(
-          IDS_OMNIBOX_SELECTED_KEYWORD_ASK_TEXT, names.short_name);
-    } else if (is_extension_keyword) {
-      names.full_name = names.short_name;
-    } else {
-      names.full_name = l10n_util::GetStringFUTF16(IDS_OMNIBOX_KEYWORD_TEXT_MD,
-                                                   names.short_name);
-    }
+    names.full_name = names.short_name;
   }
   return names;
 }
@@ -178,7 +170,7 @@ const std::u16string& SelectedKeywordVie
 
 int SelectedKeywordView::GetExtraInternalSpacing() const {
   // Align the label text with the suggestion text.
-  return 14;
+  return 3;
 }
 
 void SelectedKeywordView::SetLabelForCurrentWidth() {
