--- a/chrome/browser/ui/views/location_bar/selected_keyword_view.cc
+++ b/chrome/browser/ui/views/location_bar/selected_keyword_view.cc
@@ -58,8 +58,7 @@ SelectedKeywordView::GetKeywordLabelName
       names.full_name =
           l10n_util::GetStringUTF16(IDS_STARTER_PACK_PAGE_KEYWORD_TEXT);
     } else {
-      names.full_name = l10n_util::GetStringFUTF16(IDS_OMNIBOX_KEYWORD_TEXT_MD,
-                                                   names.short_name);
+      names.full_name = names.short_name;
     }
   }
   return names;
@@ -186,7 +185,7 @@ const std::u16string& SelectedKeywordVie
 
 int SelectedKeywordView::GetExtraInternalSpacing() const {
   // Align the label text with the suggestion text.
-  return 14;
+  return 3;
 }
 
 void SelectedKeywordView::SetLabelForCurrentWidth() {
