--- a/chrome/browser/ui/window_sizer/window_sizer.cc
+++ b/chrome/browser/ui/window_sizer/window_sizer.cc
@@ -342,9 +342,11 @@ gfx::Rect WindowSizer::GetDefaultWindowB
         static_cast<int>(work_area.width() / 2. - 1.5 * kWindowTilePixels);
   }
 #endif  // !BUILDFLAG(IS_MAC)
-  return gfx::Rect(kWindowTilePixels + work_area.x(),
-                   kWindowTilePixels + work_area.y(), default_width,
-                   default_height);
+
+  int center_x = (work_area.width() - default_width) / 2;
+  int center_y = (work_area.height() - default_height) / 2;
+
+  return gfx::Rect(center_x, center_y, default_width, default_height);
 }
 
 void WindowSizer::AdjustBoundsToBeVisibleOnDisplay(
