--- a/chrome/browser/ui/views/extensions/extension_installed_bubble_view.cc
+++ b/chrome/browser/ui/views/extensions/extension_installed_bubble_view.cc
@@ -203,22 +203,6 @@ void ExtensionInstalledBubbleView::Init(
   layout->set_cross_axis_alignment(
       views::BoxLayout::CrossAxisAlignment::kStart);
   SetLayoutManager(std::move(layout));
-
-  if (model_->show_how_to_use()) {
-    AddChildViewRaw(CreateLabel(model_->GetHowToUseText()));
-  }
-
-  if (model_->show_key_binding()) {
-    auto* manage_shortcut = AddChildView(std::make_unique<views::Link>(
-        l10n_util::GetStringUTF16(IDS_EXTENSION_INSTALLED_MANAGE_SHORTCUTS)));
-    manage_shortcut->SetCallback(base::BindRepeating(
-        &ExtensionInstalledBubbleView::LinkClicked, base::Unretained(this)));
-  }
-
-  if (model_->show_how_to_manage()) {
-    AddChildView(CreateLabel(
-        l10n_util::GetStringUTF16(IDS_EXTENSION_INSTALLED_MANAGE_INFO)));
-  }
 }
 
 void ExtensionInstalledBubbleView::LinkClicked() {
