--- a/chrome/browser/ui/views/location_bar/icon_label_bubble_view.cc
+++ b/chrome/browser/ui/views/location_bar/icon_label_bubble_view.cc
@@ -544,7 +544,7 @@ gfx::RoundedCornersF IconLabelBubbleView::GetCornerRadii() const {
   if (radii_.has_value()) {
     return radii_.value();
   }
-  return gfx::RoundedCornersF(GetPreferredSize().height() / 2);
+  return gfx::RoundedCornersF(GetPreferredSize().height() / 6);
 }
 
 void IconLabelBubbleView::SetCornerRadii(const gfx::RoundedCornersF& radii) {
--- a/chrome/browser/ui/views/location_bar/location_bar_view.cc
+++ b/chrome/browser/ui/views/location_bar/location_bar_view.cc
@@ -476,7 +476,7 @@ bool LocationBarView::IsInitialized() const {
 
 int LocationBarView::GetBorderRadius() const {
   return ChromeLayoutProvider::Get()->GetCornerRadiusMetric(
-      views::Emphasis::kMaximum, size());
+      views::Emphasis::kHigh, size());
 }
 
 std::unique_ptr<views::Background> LocationBarView::CreateRoundRectBackground(
