--- a/chrome/browser/ui/startup/first_run_service.cc
+++ b/chrome/browser/ui/startup/first_run_service.cc
@@ -23,6 +23,7 @@
 #include "chrome/browser/sync/sync_service_factory.h"
 #include "chrome/browser/ui/profiles/profile_customization_util.h"
 #include "chrome/browser/ui/profiles/profile_picker.h"
+#include "chrome/browser/themes/theme_syncable_service.h"
 #include "chrome/common/chrome_switches.h"
 #include "chrome/common/pref_names.h"
 #include "components/prefs/pref_registry_simple.h"
@@ -204,6 +205,9 @@ void FirstRunService::OnFirstRunHasExite
 void FirstRunService::FinishFirstRun(FinishedReason reason) {
   SetFirstRunFinished(reason);
 
+  profile_->GetPrefs()->SetBoolean(GetThemePrefNameInMigration(
+    ThemePrefInMigration::kGrayscaleThemeEnabled), true);
+
   // If the reason is `FinishedReason::kForceSignin` the profile is already
   // signed in and finalized. It should not finish the setup again.
   if (identity_manager_->HasPrimaryAccount(signin::ConsentLevel::kSignin) &&
