--- a/components/os_crypt/sync/keychain_password_mac.mm
+++ b/components/os_crypt/sync/keychain_password_mac.mm
@@ -28,13 +28,8 @@
 // These two strings ARE indeed user facing.  But they are used to access
 // the encryption keyword.  So as to not lose encrypted data when system
 // locale changes we DO NOT LOCALIZE.
-#if BUILDFLAG(GOOGLE_CHROME_BRANDING)
-const char kDefaultServiceName[] = "Chrome Safe Storage";
-const char kDefaultAccountName[] = "Chrome";
-#else
-const char kDefaultServiceName[] = "Chromium Safe Storage";
-const char kDefaultAccountName[] = "Chromium";
-#endif
+const char kDefaultServiceName[] = "Helium Storage Key";
+const char kDefaultAccountName[] = "Helium";
 
 // Generates a random password and adds it to the Keychain.  The added password
 // is returned from the function.  If an error occurs, an empty password is
