--- a/chrome/app/chrome_dll.rc
+++ b/chrome/app/chrome_dll.rc
@@ -49,7 +49,7 @@ BEGIN
     "W",            IDC_CLOSE_WINDOW,           VIRTKEY, CONTROL, SHIFT
     "I",            IDC_DEV_TOOLS,              VIRTKEY, CONTROL, SHIFT
     "J",            IDC_DEV_TOOLS_CONSOLE,      VIRTKEY, CONTROL, SHIFT
-    "C",            IDC_DEV_TOOLS_INSPECT,      VIRTKEY, CONTROL, SHIFT
+    "C",            IDC_COPY_URL,               VIRTKEY, CONTROL, SHIFT
     VK_F12,         IDC_DEV_TOOLS_TOGGLE,       VIRTKEY
     "F",            IDC_FIND,                   VIRTKEY, CONTROL
     "G",            IDC_FIND_NEXT,              VIRTKEY, CONTROL
--- a/chrome/browser/global_keyboard_shortcuts_mac.mm
+++ b/chrome/browser/global_keyboard_shortcuts_mac.mm
@@ -144,7 +144,7 @@ const std::vector<KeyboardShortcutData>&
 
       {true,  true,  false, false, kVK_ANSI_M,            IDC_SHOW_AVATAR_MENU},
       {true,  false, false, true,  kVK_ANSI_L,            IDC_SHOW_DOWNLOADS},
-      {true,  true,  false, false, kVK_ANSI_C,            IDC_DEV_TOOLS_INSPECT},
+      {true,  true,  false, false, kVK_ANSI_C,            IDC_COPY_URL},
       {true,  false, false, true,  kVK_ANSI_C,            IDC_DEV_TOOLS_INSPECT},
       {true,  false, false, true,  kVK_DownArrow,         IDC_FOCUS_NEXT_PANE},
       {true,  false, false, true,  kVK_UpArrow,           IDC_FOCUS_PREVIOUS_PANE},
--- a/chrome/browser/ui/cocoa/accelerators_cocoa.mm
+++ b/chrome/browser/ui/cocoa/accelerators_cocoa.mm
@@ -37,7 +37,7 @@ const struct AcceleratorMapping {
     {IDC_CUT, ui::EF_COMMAND_DOWN, ui::VKEY_X},
     {IDC_DEV_TOOLS, ui::EF_COMMAND_DOWN | ui::EF_ALT_DOWN, ui::VKEY_I},
     {IDC_DEV_TOOLS_CONSOLE, ui::EF_COMMAND_DOWN | ui::EF_ALT_DOWN, ui::VKEY_J},
-    {IDC_DEV_TOOLS_INSPECT, ui::EF_COMMAND_DOWN | ui::EF_ALT_DOWN, ui::VKEY_C},
+    {IDC_COPY_URL, ui::EF_COMMAND_DOWN | ui::EF_SHIFT_DOWN, ui::VKEY_C},
     {IDC_FIND, ui::EF_COMMAND_DOWN, ui::VKEY_F},
     {IDC_NEW_INCOGNITO_WINDOW, ui::EF_COMMAND_DOWN | ui::EF_SHIFT_DOWN,
      ui::VKEY_N},
--- a/chrome/browser/ui/views/accelerator_table.cc
+++ b/chrome/browser/ui/views/accelerator_table.cc
@@ -257,7 +257,7 @@ const AcceleratorMapping kDevToolsAccele
     {ui::VKEY_J, ui::EF_SHIFT_DOWN | ui::EF_CONTROL_DOWN,
      IDC_DEV_TOOLS_CONSOLE},
     {ui::VKEY_C, ui::EF_SHIFT_DOWN | ui::EF_CONTROL_DOWN,
-     IDC_DEV_TOOLS_INSPECT},
+     IDC_COPY_URL},
     {ui::VKEY_U, ui::EF_CONTROL_DOWN, IDC_VIEW_SOURCE},
 #endif  // !BUILDFLAG(IS_MAC)
 };
--- a/chrome/browser/ui/cocoa/main_menu_builder.mm
+++ b/chrome/browser/ui/cocoa/main_menu_builder.mm
@@ -177,6 +177,8 @@ NSMenuItem* BuildEditMenu(NSApplication*
               Item(IDS_COPY_MAC)
                   .tag(IDC_CONTENT_CONTEXT_COPY)
                   .action(@selector(copy:)),
+              Item(IDS_COPY_URL)
+                  .command_id(IDC_COPY_URL),
               Item(IDS_PASTE_MAC)
                   .tag(IDC_CONTENT_CONTEXT_PASTE)
                   .action(@selector(paste:)),
