--- a/chrome/common/chrome_features.cc
+++ b/chrome/common/chrome_features.cc
@@ -725,7 +725,7 @@ BASE_FEATURE(kHttpsFirstBalancedMode,
 // possible.
 BASE_FEATURE(kHttpsFirstBalancedModeAutoEnable,
              "HttpsFirstBalancedModeAutoEnable",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 // Enables a dialog-based UI for HTTPS-First Mode.
 BASE_FEATURE(kHttpsFirstDialogUi,
