--- a/components/policy/core/common/helium_opinionated_policy_provider.cc
+++ b/components/policy/core/common/helium_opinionated_policy_provider.cc
@@ -22,6 +22,9 @@ void HopProvider::RefreshPolicies(Policy
   PolicyBundle bundle;
   PolicyMap& policy_map = bundle.Get(PolicyNamespace(POLICY_DOMAIN_CHROME, std::string()));
 
+  Set(policy_map, policy::key::kPasswordManagerEnabled, POLICY_LEVEL_MANDATORY, base::Value(false));
+  Set(policy_map, policy::key::kPasswordManagerPasskeysEnabled, POLICY_LEVEL_MANDATORY, base::Value(false));
+
   UpdatePolicy(std::move(bundle));
 }
 
