--- a/components/policy/core/common/helium_opinionated_policy_provider.cc
+++ b/components/policy/core/common/helium_opinionated_policy_provider.cc
@@ -25,6 +25,11 @@ void HopProvider::RefreshPolicies(Policy
   Set(policy_map, policy::key::kPasswordManagerEnabled, POLICY_LEVEL_MANDATORY, base::Value(false));
   Set(policy_map, policy::key::kPasswordManagerPasskeysEnabled, POLICY_LEVEL_MANDATORY, base::Value(false));
 
+  base::Value::List preinstalled_extensions;
+  preinstalled_extensions.Append("cjpalhdlnbpafiamejdnhcphjbkeiagm"); // uBlock Origin
+  Set(policy_map, key::kExtensionInstallForcelist, POLICY_LEVEL_RECOMMENDED,
+    base::Value(std::move(preinstalled_extensions)));
+
   UpdatePolicy(std::move(bundle));
 }
 
--- a/chrome/browser/extensions/extension_management.cc
+++ b/chrome/browser/extensions/extension_management.cc
@@ -587,7 +587,7 @@ void ExtensionManagement::Refresh() {
   const base::Value::List* denied_list_pref =
       LoadListPreference(pref_names::kInstallDenyList, false);
   const base::Value::Dict* forced_list_pref =
-      LoadDictPreference(pref_names::kInstallForceList, true);
+      LoadDictPreference(pref_names::kInstallForceList, false);
   const base::Value::List* install_sources_pref =
       LoadListPreference(pref_names::kAllowedInstallSites, true);
   const base::Value::List* allowed_types_pref =
