--- a/components/policy/core/common/helium_opinionated_policy_provider.cc
+++ b/components/policy/core/common/helium_opinionated_policy_provider.cc
@@ -25,6 +25,11 @@ void HopProvider::RefreshPolicies(Policy
   Set(policy_map, policy::key::kPasswordManagerEnabled, POLICY_LEVEL_MANDATORY, base::Value(false));
   Set(policy_map, policy::key::kPasswordManagerPasskeysEnabled, POLICY_LEVEL_MANDATORY, base::Value(false));
 
+  base::Value::List preinstalled_extensions;
+  preinstalled_extensions.Append("cjpalhdlnbpafiamejdnhcphjbkeiagm"); // uBlock Origin
+  Set(policy_map, key::kExtensionInstallForcelist, POLICY_LEVEL_RECOMMENDED,
+    base::Value(std::move(preinstalled_extensions)));
+
   UpdatePolicy(std::move(bundle));
 }
 
--- a/components/policy/core/common/policy_loader_common.cc
+++ b/components/policy/core/common/policy_loader_common.cc
@@ -24,9 +24,9 @@ namespace {
 // implementation. And the Chrome one can be moved to c/b/policy.
 // From extensions/common/extension_urls.cc
 const char kChromeWebstoreUpdateURL[] =
-    "https://clients2.google.com/service/update2/crx";
+    "https://services.helium.computer/ext/webstore";
 const char16_t kChromeWebstoreUpdateURL16[] =
-    u"https://clients2.google.com/service/update2/crx";
+    u"https://services.helium.computer/ext/webstore";
 
 // From chrome/browser/extensions/extension_management_constants.cc
 const char kWildcard[] = "*";
