--- a/chrome/browser/search_engines/template_url_service_factory.cc
+++ b/chrome/browser/search_engines/template_url_service_factory.cc
@@ -28,6 +28,7 @@
 #include "components/search_engines/enterprise/enterprise_search_manager.h"
 #include "components/search_engines/search_engines_pref_names.h"
 #include "components/search_engines/template_url_service.h"
+#include "content/public/browser/storage_partition.h"
 #include "rlz/buildflags/buildflags.h"
 
 #if BUILDFLAG(IS_CHROMEOS)
@@ -65,6 +66,7 @@ std::unique_ptr<KeyedService> TemplateUR
   Profile* profile = Profile::FromBrowserContext(context);
   return std::make_unique<TemplateURLService>(
       CHECK_DEREF(profile->GetPrefs()),
+      context->GetDefaultStoragePartition()->GetURLLoaderFactoryForBrowserProcess(),
       CHECK_DEREF(
           search_engines::SearchEngineChoiceServiceFactory::GetForProfile(
               profile)),
--- a/components/search_engines/template_url_data_util.h
+++ b/components/search_engines/template_url_data_util.h
@@ -8,6 +8,7 @@
 #include <memory>
 
 #include "base/values.h"
+#include "components/search_engines/template_url_bang_manager.h"
 
 namespace TemplateURLPrepopulateData {
 struct PrepopulatedEngine;
@@ -40,4 +41,5 @@ std::unique_ptr<TemplateURLData> Templat
 std::unique_ptr<TemplateURLData> TemplateURLDataFromStarterPackEngine(
     const TemplateURLStarterPackData::StarterPackEngine& engine);
 
+TemplateURLData TemplateURLDataFromBang(const Bangs::Bang& bang);
 #endif  // COMPONENTS_SEARCH_ENGINES_TEMPLATE_URL_DATA_UTIL_H_
--- a/components/search_engines/BUILD.gn
+++ b/components/search_engines/BUILD.gn
@@ -36,6 +36,8 @@ static_library("search_engines") {
     "search_terms_data.h",
     "template_url.cc",
     "template_url.h",
+    "template_url_bang_manager.cc",
+    "template_url_bang_manager.h",
     "template_url_data.cc",
     "template_url_data.h",
     "template_url_data_util.cc",
@@ -79,6 +81,7 @@ static_library("search_engines") {
     "//components/country_codes",
     "//components/crash/core/common:crash_key",
     "//components/database_utils",
+    "//components/helium_services",
     "//components/infobars/core",
     "//components/lens:lens",
     "//components/lens/proto/server:proto",
--- a/components/search_engines/template_url_data_util.cc
+++ b/components/search_engines/template_url_data_util.cc
@@ -13,6 +13,7 @@
 #include "components/search_engines/default_search_manager.h"
 #include "components/search_engines/template_url_data.h"
 #include "components/search_engines/template_url_starter_pack_data.h"
+#include "components/search_engines/template_url_bang_manager.h"
 #include "third_party/search_engines_data/resources/definitions/prepopulated_engines.h"
 #include "ui/base/l10n/l10n_util.h"
 #include "url/gurl.h"
@@ -507,3 +508,20 @@ std::unique_ptr<TemplateURLData> Templat
 
   return turl;
 }
+
+TemplateURLData TemplateURLDataFromBang(const Bangs::Bang& bang) {
+  TemplateURLData turl;
+  turl.SetShortName(base::UTF8ToUTF16(bang.name));
+  turl.SetKeyword(u"!" + base::UTF8ToUTF16(bang.bang));
+  turl.SetURL(bang.template_url);
+
+  GURL favicon_url = GURL(bang.template_url).GetWithEmptyPath();
+  favicon_url.path() = "/favicon.ico";
+  turl.favicon_url = std::move(favicon_url);
+  turl.bang_id = 1 + bang.id;
+
+  turl.safe_for_autoreplace = false;
+  turl.GenerateSyncGUID();
+  turl.is_active = TemplateURLData::ActiveStatus::kTrue;
+  return turl;
+}
--- a/components/search_engines/template_url_service.cc
+++ b/components/search_engines/template_url_service.cc
@@ -43,6 +43,8 @@
 #include "base/strings/utf_string_conversions.h"
 #include "build/build_config.h"
 #include "components/country_codes/country_codes.h"
+#include "components/helium_services/pref_names.h"
+#include "components/helium_services/helium_services_helpers.h"
 #include "components/omnibox/common/omnibox_features.h"
 #include "components/pref_registry/pref_registry_syncable.h"
 #include "components/prefs/pref_service.h"
@@ -56,7 +58,9 @@
 #include "components/search_engines/search_engines_switches.h"
 #include "components/search_engines/search_terms_data.h"
 #include "components/search_engines/template_url.h"
+#include "components/search_engines/template_url_bang_manager.h"
 #include "components/search_engines/template_url_data.h"
+#include "components/search_engines/template_url_data_util.h"
 #include "components/search_engines/template_url_prepopulate_data.h"
 #include "components/search_engines/template_url_prepopulate_data_resolver.h"
 #include "components/search_engines/template_url_service_client.h"
@@ -70,6 +74,7 @@
 #include "components/sync/protocol/search_engine_specifics.pb.h"
 #include "components/url_formatter/url_fixer.h"
 #include "net/base/registry_controlled_domains/registry_controlled_domain.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
 #include "url/gurl.h"
 #include "url/origin.h"
 #include "url/url_util.h"
@@ -470,6 +475,7 @@ class TemplateURLService::PreLoadingProv
 // TemplateURLService ---------------------------------------------------------
 TemplateURLService::TemplateURLService(
     PrefService& prefs,
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory,
     search_engines::SearchEngineChoiceService& search_engine_choice_service,
     TemplateURLPrepopulateData::Resolver& prepopulate_data_resolver,
     std::unique_ptr<SearchTermsData> search_terms_data,
@@ -482,6 +488,8 @@ TemplateURLService::TemplateURLService(
       prepopulate_data_resolver_(prepopulate_data_resolver),
       search_terms_data_(std::move(search_terms_data)),
       web_data_service_(web_data_service),
+      url_loader_factory_(std::move(url_loader_factory)),
+      bang_manager_(Bangs::BangManager::GetInstance()),
       client_(std::move(client)),
       dsp_change_callback_(dsp_change_callback),
       pre_loading_providers_(std::make_unique<PreLoadingProviders>()),
@@ -498,11 +506,13 @@ TemplateURLService::TemplateURLService(
 
 TemplateURLService::TemplateURLService(
     PrefService& prefs,
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory,
     search_engines::SearchEngineChoiceService& search_engine_choice_service,
     TemplateURLPrepopulateData::Resolver& prepopulate_data_resolver,
     base::span<const TemplateURLService::Initializer> initializers)
     : TemplateURLService(
           prefs,
+          url_loader_factory,
           search_engine_choice_service,
           prepopulate_data_resolver,
           /*search_terms_data=*/std::make_unique<SearchTermsData>(),
@@ -603,6 +613,10 @@ bool TemplateURLService::ShowInActivesLi
 }
 
 bool TemplateURLService::HiddenFromLists(const TemplateURL* t_url) const {
+  if (t_url->bang_id()) {
+    return true;
+  }
+
   switch (t_url->policy_origin()) {
     case TemplateURLData::PolicyOrigin::kNoPolicy:
       // Hide if the preferred search engine for the keyword is created by
@@ -1445,11 +1459,37 @@ void TemplateURLService::Load() {
 
   if (web_data_service_) {
     load_handle_ = web_data_service_->GetKeywords(this);
+    LoadBangs();
   } else {
     ChangeToLoadedState();
   }
 }
 
+void TemplateURLService::LoadBangs() {
+  LOG(ERROR) << "LoadBangs() called";
+  bang_manager_->LoadBangs(url_loader_factory_, prefs_.get(),
+    base::BindOnce(&TemplateURLService::BangsLoadedCallback, base::Unretained(this)));
+}
+
+void TemplateURLService::BangsLoadedCallback() {
+  LOG(INFO) << "BangsLoadedCallback() called";
+
+  for (const auto& bang : bang_manager_->GetBangs()) {
+    auto template_data = TemplateURLDataFromBang(bang);
+
+    if (!GetTemplateURLForKeyword(template_data.keyword())) {
+      template_data.id = ++next_id_;
+
+      auto t_url = std::make_unique<TemplateURL>(
+        std::move(template_data), TemplateURL::Type::LOCAL);
+      auto t_ptr = t_url.get();
+
+      template_urls_.push_back(std::move(t_url));
+      AddToMaps(t_ptr);
+    }
+  }
+}
+
 base::CallbackListSubscription TemplateURLService::RegisterOnLoadedCallback(
     base::OnceClosure callback) {
   return loaded_ ? base::CallbackListSubscription()
@@ -2306,6 +2346,13 @@ void TemplateURLService::Init() {
             base::Unretained(this)));
   }
 
+  helium::ConfigurePrefChangeRegistrarFor(
+    prefs::kHeliumBangsEnabled, pref_change_registrar_,
+    base::BindRepeating(
+        &TemplateURLService::LoadBangs,
+        base::Unretained(this))
+  );
+
   DefaultSearchManager::Source source = DefaultSearchManager::FROM_USER;
   const TemplateURLData* dse =
       default_search_manager_.GetDefaultSearchEngine(&source);
--- a/components/search_engines/template_url_service.h
+++ b/components/search_engines/template_url_service.h
@@ -34,10 +34,12 @@
 #include "components/search_engines/search_host_to_urls_map.h"
 #include "components/search_engines/search_terms_data.h"
 #include "components/search_engines/template_url.h"
+#include "components/search_engines/template_url_bang_manager.h"
 #include "components/sync/model/sync_change.h"
 #include "components/sync/model/syncable_service.h"
 #include "components/sync/protocol/search_engine_specifics.pb.h"
 #include "components/webdata/common/web_data_service_consumer.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
 #if BUILDFLAG(IS_ANDROID)
 #include "base/android/scoped_java_ref.h"
 #endif
@@ -142,6 +144,7 @@ class TemplateURLService final : public
 
   TemplateURLService(
       PrefService& prefs,
+      scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory,
       search_engines::SearchEngineChoiceService& search_engine_choice_service,
       TemplateURLPrepopulateData::Resolver& prepopulate_data_resolver,
       std::unique_ptr<SearchTermsData> search_terms_data,
@@ -153,6 +156,7 @@ class TemplateURLService final : public
   // some template URL data.
   explicit TemplateURLService(
       PrefService& prefs,
+      scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory,
       search_engines::SearchEngineChoiceService& search_engine_choice_service,
       TemplateURLPrepopulateData::Resolver& prepopulate_data_resolver,
       base::span<const TemplateURLService::Initializer> initializers = {});
@@ -475,6 +479,8 @@ class TemplateURLService final : public
   // OnTemplateURLServiceChanged.
   void Load();
 
+  void LoadBangs();
+
   // Registers a callback to be called when the service has loaded.
   //
   // If the service has already loaded, this function does nothing.
@@ -697,6 +703,8 @@ class TemplateURLService final : public
 
   void Init();
 
+  void BangsLoadedCallback();
+
   // Simulate a loaded `TemplateURLService`.
   void ApplyInitializersForTesting(
       base::span<const TemplateURLService::Initializer> initializers);
@@ -903,6 +911,10 @@ class TemplateURLService final : public
   // Service used to store entries.
   scoped_refptr<KeywordWebDataService> web_data_service_;
 
+  scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory_;
+
+  Bangs::BangManager* bang_manager_;
+
   std::unique_ptr<TemplateURLServiceClient> client_;
 
   // This closure is run when the default search provider is set to Google.
--- /dev/null
+++ b/components/search_engines/template_url_bang_manager.cc
@@ -0,0 +1,174 @@
+// Copyright 2025 The Helium Authors
+// You can use, redistribute, and/or modify this source code under
+// the terms of the GPL-3.0 license that can be found in the LICENSE file.
+
+#include "components/search_engines/template_url_bang_manager.h"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/functional/bind.h"
+#include "base/functional/callback.h"
+#include "base/memory/weak_ptr.h"
+#include "base/memory/singleton.h"
+#include "base/json/json_reader.h"
+#include "components/helium_services/helium_services_helpers.h"
+#include "net/base/load_flags.h"
+#include "net/http/http_status_code.h"
+#include "services/network/public/cpp/resource_request.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "services/network/public/mojom/url_response_head.mojom.h"
+#include "url/gurl.h"
+
+namespace Bangs {
+
+const int kMaxFetchRetries = 3;
+const int kFetchTimeoutSeconds = 30;
+
+constexpr net::NetworkTrafficAnnotationTag kCustomizationDocumentNetworkTag =
+    net::DefineNetworkTrafficAnnotation("native_bangs",
+                                        R"(
+        semantics {
+            sender: "Native bang loader"
+            description:
+            "Get data for native bangs."
+            trigger:
+            "Triggered when the browser is opened. "
+            data: "None."
+            destination: GOOGLE_OWNED_SERVICE
+        }
+        policy {
+            cookies_allowed: NO
+        })");
+
+
+BangManager* BangManager::GetInstance() {
+    return base::Singleton<
+    BangManager,
+        base::DefaultSingletonTraits<BangManager>>::get();
+}
+
+void BangManager::LoadBangs(
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory,
+    PrefService& prefs,
+    base::OnceClosure callback) {
+    if (bangs_.empty() && !load_pending_) {
+        load_pending_ = true;
+        DoStartFileFetch(url_loader_factory, prefs);
+    } else if (!load_pending_) {
+        std::move(callback).Run();
+        return;
+    }
+
+    callbacks_.push_back(std::move(callback));
+}
+
+void BangManager::DoStartFileFetch(
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory,
+    PrefService& prefs
+) {
+    auto request = std::make_unique<network::ResourceRequest>();
+    request->url = helium::GetServicesBaseURL(prefs).Resolve(kBangsPrefix);
+
+    request->load_flags = helium::ShouldFetchBangs(prefs)
+                            ? net::LOAD_NORMAL
+                            : net::LOAD_ONLY_FROM_CACHE;
+
+    request->credentials_mode = network::mojom::CredentialsMode::kOmit;
+    request->headers.SetHeader("Accept", "application/json");
+
+    url_loader_ = network::SimpleURLLoader::Create(
+        std::move(request), kCustomizationDocumentNetworkTag);
+
+    url_loader_->SetTimeoutDuration(base::Seconds(kFetchTimeoutSeconds));
+    url_loader_->SetRetryOptions(
+        kMaxFetchRetries, network::SimpleURLLoader::RETRY_ON_NETWORK_CHANGE);
+
+    url_loader_->DownloadToString(
+        url_loader_factory.get(),
+        base::BindOnce(&BangManager::OnSimpleLoaderComplete, base::Unretained(this)),
+        5e6 /* 5 MB */
+    );
+}
+
+void BangManager::OnSimpleLoaderComplete(
+    std::unique_ptr<std::string> response_body) {
+    int response_code = -1;
+    std::string mime_type;
+
+    if (url_loader_->ResponseInfo() && url_loader_->ResponseInfo()->headers) {
+        response_code = url_loader_->ResponseInfo()->headers->response_code();
+        url_loader_->ResponseInfo()->headers->GetMimeType(&mime_type);
+    }
+
+    if (response_body && mime_type == "application/json") {
+        BangManager::LoadManifestFromString(*response_body);
+    } else {
+        LOG(ERROR) << "Fetching bangs failed:"
+                   << " response code = " << response_code;
+    }
+
+    load_pending_ = false;
+}
+
+void BangManager::LoadManifestFromString(const std::string& manifest) {
+    auto parsed = base::JSONReader::ReadAndReturnValueWithError(
+        manifest,
+        base::JSON_ALLOW_COMMENTS
+    );
+
+    if (!parsed.has_value()) {
+        LOG(ERROR) << parsed.error().message;
+        return;
+    } else if (!parsed->is_list()) {
+        LOG(ERROR) << "received invalid bang data - not an array";
+        return;
+    }
+
+    auto list = std::move(*parsed).TakeList();
+    bangs_.reserve(list.size());
+
+    size_t index = 0;
+
+    for (auto& bang : list) {
+        if (!bang.is_dict()) {
+            continue;
+        }
+
+        auto dict = std::move(bang).TakeDict();
+        auto bname = dict.FindString("s");
+        auto bbang = dict.FindString("t");
+        auto burl_template = dict.FindString("u");
+        if (!bname || !bbang || !burl_template) {
+            continue;
+        }
+
+        auto b_gurl = GURL(*burl_template);
+        if (!b_gurl.is_valid() || !b_gurl.SchemeIsHTTPOrHTTPS()) {
+            LOG(INFO) << "invalid bang url " << *burl_template;
+            continue;
+        }
+
+        // We need to pass the URL here as-is, because otherwise URLs
+        // which have the template in the path will have it %-encoded,
+        // which will break everything.
+        bangs_.emplace_back(index++, *bname, *bbang, *burl_template);
+    }
+
+    LOG(INFO) << bangs_.size() << " bangs loaded";
+    if (!callbacks_.empty()) {
+        auto callbacks = std::move(callbacks_);
+        for (auto& cb : callbacks) {
+            std::move(cb).Run();
+        }
+    }
+}
+
+const std::vector<Bang>& BangManager::GetBangs() {
+    return bangs_;
+}
+
+}
+
--- /dev/null
+++ b/components/search_engines/template_url_bang_manager.h
@@ -0,0 +1,65 @@
+// Copyright 2025 The Helium Authors
+// You can use, redistribute, and/or modify this source code under
+// the terms of the GPL-3.0 license that can be found in the LICENSE file.
+
+#ifndef COMPONENTS_SEARCH_ENGINES_BANG_PROVIDER_H_
+#define COMPONENTS_SEARCH_ENGINES_BANG_PROVIDER_H_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/memory/weak_ptr.h"
+#include "base/functional/callback.h"
+#include "components/prefs/pref_service.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "url/gurl.h"
+
+namespace Bangs {
+    // Template URL where to fetch bangs data from.
+    static constexpr char kBangsPrefix[] = "/bangs.json";
+
+    struct Bang {
+        size_t id;
+        std::string name;
+        std::string bang;
+        std::string template_url;
+    };
+
+    class BangManager {
+    public:
+        static BangManager* GetInstance();
+
+        BangManager() = default;
+        BangManager(const BangManager&) = delete;
+        BangManager& operator=(const BangManager&) = delete;
+
+        void LoadBangs(
+            scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory,
+            PrefService& prefs,
+            base::OnceClosure callback
+        );
+        const std::vector<Bang>& GetBangs();
+
+    private:
+        void DoStartFileFetch(
+            scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory,
+            PrefService& prefs
+        );
+        void OnSimpleLoaderComplete(std::unique_ptr<std::string> response_body);
+        void LoadManifestFromString(const std::string& manifest);
+
+        // Weak factory for callbacks.
+        base::WeakPtrFactory<BangManager> weak_ptr_factory_{this};
+        // SimpleURLLoader instance.
+        std::unique_ptr<network::SimpleURLLoader> url_loader_;
+
+        std::vector<Bang> bangs_;
+        bool load_pending_;
+        std::vector<base::OnceClosure> callbacks_;
+    };
+}
+
+#endif  // COMPONENTS_SEARCH_ENGINES_BANG_PROVIDER_H_
+
--- a/components/omnibox/browser/autocomplete_input.cc
+++ b/components/omnibox/browser/autocomplete_input.cc
@@ -743,9 +743,9 @@ bool AutocompleteInput::HasHTTPSScheme(c
 // static
 AutocompleteInput::FeaturedKeywordMode
 AutocompleteInput::GetFeaturedKeywordMode(std::u16string_view text) {
-  if (text == u"@")
+  if (text == u"@" || text == u"!")
     return FeaturedKeywordMode::kExact;
-  if (text.starts_with(u'@'))
+  if (text.starts_with(u'@') || text.starts_with(u'!'))
     return FeaturedKeywordMode::kPrefix;
   return FeaturedKeywordMode::kFalse;
 }
--- a/components/search_engines/template_url_data.cc
+++ b/components/search_engines/template_url_data.cc
@@ -27,7 +27,7 @@ namespace {
 // Returns a GUID used for sync, which is random except for built-in search
 // engines. The latter benefit from using a deterministic GUID, to make sure
 // sync doesn't incur in duplicates for prepopulated engines.
-std::string GenerateGUID(int prepopulate_id, int starter_pack_id) {
+std::string GenerateGUID(int prepopulate_id, int starter_pack_id, int bang_id) {
   // We compute a GUID deterministically given |prepopulate_id| or
   // |starter_pack_id|, using an arbitrary base GUID.
   std::string guid;
@@ -38,6 +38,9 @@ std::string GenerateGUID(int prepopulate
   } else if (starter_pack_id > 0) {
     guid = base::StringPrintf("ec205736-edd7-4022-a9a3-b431fc%06d",
                               starter_pack_id);
+  } else if (bang_id > 0) {
+    guid = base::StringPrintf("2bfdeb48-cca4-4647-8843-7bb6c8%06d",
+                              bang_id);
   } else {
     guid = base::Uuid::GenerateRandomV4().AsLowercaseString();
   }
@@ -117,7 +120,7 @@ TemplateURLData::TemplateURLData(
       created_from_play_api(false),
       usage_count(0),
       prepopulate_id(prepopulate_id),
-      sync_guid(GenerateGUID(prepopulate_id, 0)),
+      sync_guid(GenerateGUID(prepopulate_id, 0, 0)),
       preconnect_to_search_url(preconnect_to_search_url),
       prefetch_likely_navigations(prefetch_likely_navigations) {
   SetShortName(name);
@@ -183,7 +186,7 @@ std::vector<uint8_t> TemplateURLData::Ge
 }
 
 void TemplateURLData::GenerateSyncGUID() {
-  sync_guid = GenerateGUID(prepopulate_id, starter_pack_id);
+  sync_guid = GenerateGUID(prepopulate_id, starter_pack_id, bang_id);
 }
 
 size_t TemplateURLData::EstimateMemoryUsage() const {
--- a/components/search_engines/template_url_data.h
+++ b/components/search_engines/template_url_data.h
@@ -245,6 +245,9 @@ struct TemplateURLData {
   // starter_pack_id > 0.
   int starter_pack_id{0};
 
+  // This TemplateURL is a bang if bang_id > 0.
+  int bang_id{0};
+
   friend bool operator==(const TemplateURLData&,
                          const TemplateURLData&) = default;
 
--- a/components/search_engines/template_url.cc
+++ b/components/search_engines/template_url.cc
@@ -1683,6 +1683,8 @@ bool TemplateURL::IsBetterThanConflictin
         !engine->safe_for_autoreplace(),
         // Prefer engines created by Play API.
         engine->created_from_play_api(),
+        // Favor bangs over other auto-generated engines.
+        engine->bang_id() > 0,
         // Favor prepopulated engines over other auto-generated engines.
         engine->prepopulate_id() > 0,
         // Favor starter pack engines over other auto-generated engines.
--- a/components/search_engines/template_url.h
+++ b/components/search_engines/template_url.h
@@ -819,6 +819,8 @@ class TemplateURL {
 
   int starter_pack_id() const { return data().starter_pack_id; }
 
+  int bang_id() const { return data().bang_id; }
+
   const std::vector<TemplateURLRef>& url_refs() const { return url_refs_; }
   const TemplateURLRef& url_ref() const {
     // Sanity check for https://crbug.com/781703.
--- a/components/helium_services/helium_services_helpers.cc
+++ b/components/helium_services/helium_services_helpers.cc
@@ -65,6 +65,11 @@ bool ShouldAccessExtensionService(const
             prefs.GetBoolean(prefs::kHeliumExtProxyEnabled);
 }
 
+bool ShouldFetchBangs(const PrefService& prefs) {
+    return ShouldAccessServices(prefs) &&
+            prefs.GetBoolean(prefs::kHeliumBangsEnabled);
+}
+
 GURL GetExtensionUpdateURL(const PrefService& prefs) {
     if (!ShouldAccessExtensionService(prefs)) {
         return GetDummyURL();
--- a/components/helium_services/helium_services_helpers.h
+++ b/components/helium_services/helium_services_helpers.h
@@ -18,6 +18,7 @@ const char kHeliumDefaultOrigin[] =
 const char kHeliumDummyOrigin[] =
     "https://helium-services-are-disabled.qjz9zk";
 
+COMPONENT_EXPORT(HELIUM) bool ShouldFetchBangs(const PrefService& prefs);
 COMPONENT_EXPORT(HELIUM) bool ShouldAccessExtensionService(const PrefService& prefs);
 COMPONENT_EXPORT(HELIUM) GURL GetServicesBaseURL(const PrefService& prefs);
 COMPONENT_EXPORT(HELIUM) GURL GetDummyURL();
--- a/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc
+++ b/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc
@@ -1850,6 +1850,9 @@ void AddPrivacyStrings(content::WebUIDat
       {"heliumServicesToggle", IDS_SETTINGS_HELIUM_SERVICES_TOGGLE},
       {"heliumServicesToggleDescription",
         IDS_SETTINGS_HELIUM_SERVICES_TOGGLE_DESCRIPTION},
+      {"heliumBangsToggle", IDS_SETTINGS_HELIUM_SERVICES_BANGS_TOGGLE},
+      {"heliumBangsToggleDescription",
+        IDS_SETTINGS_HELIUM_SERVICES_BANGS_TOGGLE_DESCRIPTION},
       {"heliumExtProxyToggle",
         IDS_SETTINGS_HELIUM_SERVICES_EXT_PROXY_TOGGLE},
       {"heliumExtProxyToggleDescription",
--- a/chrome/app/settings_strings.grdp
+++ b/chrome/app/settings_strings.grdp
@@ -2085,6 +2085,12 @@
   <message name="IDS_SETTINGS_HELIUM_SERVICES_EXT_PROXY_TOGGLE_DESCRIPTION" desc="Description of the toggle for enabling/disabling of downloading and proxying extensions">
     When enabled, Helium will proxy extension downloads and updates to protect your privacy. When disabled, downloading and updating extensions will not work.
   </message>
+  <message name="IDS_SETTINGS_HELIUM_SERVICES_BANGS_TOGGLE" desc="Toggle for enabling/disabling of downloading bangs">
+    Allow downloading the !bang list
+  </message>
+  <message name="IDS_SETTINGS_HELIUM_SERVICES_BANGS_TOGGLE_DESCRIPTION" desc="Description of the toggle for enabling/disabling of downloading bangs">
+    When enabled, Helium will fetch our list of bangs that help you browse the Internet faster, such as !w or !gh.
+  </message>
   <message name="IDS_SETTINGS_HELIUM_SERVICES_OVERRIDE" desc="Text input for overriding the Helium services server">
     Use your own instance of Helium services
   </message>
--- a/chrome/browser/resources/settings/privacy_page/services_page.html
+++ b/chrome/browser/resources/settings/privacy_page/services_page.html
@@ -12,6 +12,11 @@
     </settings-toggle-button>
     <div class="cr-col">
         <cr-collapse id="servicesCollapse" opened="[[prefs.helium.services.enabled.value]]">
+            <settings-toggle-button id="bangToggleButton"
+                pref="{{prefs.helium.services.bangs}}"
+                label="$i18n{heliumBangsToggle}"
+                sub-label="$i18n{heliumBangsToggleDescription}">
+            </settings-toggle-button>
             <settings-toggle-button id="extProxyToggleButton"
                 pref="{{prefs.helium.services.ext_proxy}}"
                 label="$i18n{heliumExtProxyToggle}"
--- a/chrome/browser/ui/browser_ui_prefs.cc
+++ b/chrome/browser/ui/browser_ui_prefs.cc
@@ -172,6 +172,7 @@ void RegisterBrowserUserPrefs(user_prefs
 
   {
     registry->RegisterBooleanPref(prefs::kHeliumServicesEnabled, true);
+    registry->RegisterBooleanPref(prefs::kHeliumBangsEnabled, true);
     registry->RegisterBooleanPref(prefs::kHeliumExtProxyEnabled, true);
     registry->RegisterStringPref(prefs::kHeliumServicesOrigin, "");
     registry->RegisterBooleanPref(prefs::kHeliumDidOnboarding, false);
--- a/chrome/browser/extensions/api/settings_private/prefs_util.cc
+++ b/chrome/browser/extensions/api/settings_private/prefs_util.cc
@@ -344,6 +344,8 @@ const PrefsUtil::TypedPrefMap& PrefsUtil
       settings_api::PrefType::kBoolean;
   (*s_allowlist)[::prefs::kHeliumExtProxyEnabled] =
       settings_api::PrefType::kBoolean;
+  (*s_allowlist)[::prefs::kHeliumBangsEnabled] =
+      settings_api::PrefType::kBoolean;
   (*s_allowlist)[::prefs::kHeliumServicesOrigin] =
       settings_api::PrefType::kString;
 
--- a/components/helium_services/pref_names.h
+++ b/components/helium_services/pref_names.h
@@ -21,6 +21,9 @@ inline constexpr char kHeliumDidOnboardi
 inline constexpr char kHeliumExtProxyEnabled[] =
     "helium.services.ext_proxy";
 
+inline constexpr char kHeliumBangsEnabled[] =
+    "helium.services.bangs";
+
 }  // namespace prefs
 
 #endif  // COMPONENTS_HELIUM_SERVICES_PREF_NAMES_H_
