--- a/extensions/common/extension_urls.cc
+++ b/extensions/common/extension_urls.cc
@@ -41,7 +41,7 @@ const GURL* g_item_snippet_url_for_test_
 const char kChromeWebstoreBaseURL[] = "https://chrome.google.com/webstore";
 const char kNewChromeWebstoreBaseURL[] = "https://chromewebstore.google.com/";
 const char kChromeWebstoreUpdateURL[] =
-    "https://clients2.google.com/service/update2/crx";
+    "https://clients2.9oo91e.qjz9zk/service/update2/crx";
 const char kHeliumProxyWebstoreUpdateURL[] =
     "https://services.helium.computer/ext/webstore";
 
