--- a/chrome/browser/ui/ui_features.cc
+++ b/chrome/browser/ui/ui_features.cc
@@ -165,12 +165,7 @@ BASE_FEATURE(kTabGroupsCollapseFreezing,
 // https://crbug.com/928954
 BASE_FEATURE(kTabHoverCardImages,
              "TabHoverCardImages",
-#if BUILDFLAG(IS_MAC)
-             base::FEATURE_DISABLED_BY_DEFAULT
-#else
-             base::FEATURE_ENABLED_BY_DEFAULT
-#endif
-);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 const char kTabHoverCardImagesNotReadyDelayParameterName[] =
     "page_not_ready_delay";
