--- a/extensions/common/extension_urls.cc
+++ b/extensions/common/extension_urls.cc
@@ -39,10 +39,10 @@ const GURL* g_item_snippet_url_for_test_
 
 }  // namespace
 
-const char kChromeWebstoreBaseURL[] = "trk:09:https://chrome.google.com/webstore";
-const char kNewChromeWebstoreBaseURL[] = "trk:08:https://chromewebstore.google.com/";
+const char kChromeWebstoreBaseURL[] = "https://chrome.google.com/webstore";
+const char kNewChromeWebstoreBaseURL[] = "https://chromewebstore.google.com/";
 const char kChromeWebstoreUpdateURL[] =
-    "trk:05:https://clients2.google.com/service/update2/crx";
+    "https://clients2.google.com/service/update2/crx";
 
 const char kAppMenuUtmSource[] = "ext_app_menu";
 const char kExtensionsMenuUtmSource[] = "ext_extensions_menu";
