--- a/chrome/browser/ui/views/toolbar/app_menu.cc
+++ b/chrome/browser/ui/views/toolbar/app_menu.cc
@@ -250,7 +250,7 @@ class InMenuButtonBackground : public vi
       constexpr int kCircularButtonSize = 28;
       bounds_rect.ClampToCenteredSize(
           gfx::Size(kCircularButtonSize, kCircularButtonSize));
-      menu_item.corner_radius = kCircularButtonSize / 2;
+      menu_item.corner_radius = 8;
     }
     const auto* const color_provider = view->GetColorProvider();
     cc::PaintFlags flags;
@@ -1424,7 +1424,7 @@ void AppMenu::PopulateMenu(MenuItemView*
     // Helper method that adds a background to a menu item.
     auto add_menu_row_background = [item](int vertical_margin,
                                           ui::ColorId background_color_id) {
-      constexpr int kBackgroundCornerRadius = 12;
+      constexpr int kBackgroundCornerRadius = 8;
       item->set_vertical_margin(vertical_margin);
       item->SetMenuItemBackground(MenuItemView::MenuItemBackground(
           background_color_id, kBackgroundCornerRadius));
@@ -1471,7 +1471,7 @@ void AppMenu::PopulateMenu(MenuItemView*
       case IDC_SET_BROWSER_AS_DEFAULT: {
         // Only highlight the default browser item when it is first in the
         // AppMenu.
-        if (i == 0) {
+        if ((false)) {
           add_menu_row_background(
               views::LayoutProvider::Get()->GetDistanceMetric(
                   views::DISTANCE_CONTROL_VERTICAL_TEXT_PADDING),
