--- a/chrome/browser/resources/side_panel/customize_chrome/app.ts
+++ b/chrome/browser/resources/side_panel/customize_chrome/app.ts
@@ -138,24 +138,6 @@ export class AppElement extends AppEleme
                   element.scrollIntoView({behavior: 'auto'});
                 });
 
-    this.attachedTabStateUpdatedId_ =
-        CustomizeChromeApiProxy.getInstance()
-            .callbackRouter.attachedTabStateUpdated.addListener(
-                (isSourceTabFirstPartyNtp: boolean) => {
-                  if (this.isSourceTabFirstPartyNtp_ ===
-                      isSourceTabFirstPartyNtp) {
-                    return;
-                  }
-
-                  this.isSourceTabFirstPartyNtp_ = isSourceTabFirstPartyNtp;
-
-                  // Since some pages aren't supported in non first party mode,
-                  // change the section back to the overview.
-                  if (!this.isSourceTabFirstPartyNtp_ &&
-                      !this.pageSupportedOnNonFirstPartyNtps()) {
-                    this.page_ = CustomizeChromePage.OVERVIEW;
-                  }
-                });
     this.pageHandler_.updateAttachedTabState();
 
     this.setThemeEditableId_ = CustomizeChromeApiProxy.getInstance()
@@ -290,10 +272,6 @@ export class AppElement extends AppEleme
     await this.updateComplete;
     page.focusOnBackButton();
   }
-
-  private pageSupportedOnNonFirstPartyNtps() {
-    return this.page_ === CustomizeChromePage.TOOLBAR;
-  }
 }
 
 declare global {
--- a/chrome/browser/resources/side_panel/customize_chrome/appearance.ts
+++ b/chrome/browser/resources/side_panel/customize_chrome/appearance.ts
@@ -131,12 +131,6 @@ export class AppearanceElement extends A
         });
     this.pageHandler_.updateTheme();
 
-    this.attachedTabStateUpdatedId_ =
-        CustomizeChromeApiProxy.getInstance()
-            .callbackRouter.attachedTabStateUpdated.addListener(
-                (isSourceTabFirstPartyNtp: boolean) => {
-                  this.isSourceTabFirstPartyNtp_ = isSourceTabFirstPartyNtp;
-                });
     this.pageHandler_.updateAttachedTabState();
 
     this.setThemeEditableId_ = CustomizeChromeApiProxy.getInstance()
