--- a/ui/views/layout/layout_provider.cc
+++ b/ui/views/layout/layout_provider.cc
@@ -86,7 +86,7 @@ int LayoutProvider::GetDistanceMetric(in
     case DISTANCE_BUBBLE_PREFERRED_WIDTH:
       return kSmallDialogWidth;
     case DISTANCE_BUTTON_HORIZONTAL_PADDING:
-      return 16;
+      return 14;
     case DISTANCE_BUTTON_MAX_LINKABLE_WIDTH:
       return 112;
     case DISTANCE_CLOSE_BUTTON_MARGIN:
@@ -201,19 +201,19 @@ ShapeSysTokens GetShapeSysToken(ShapeCon
   static constexpr auto shape_token_map =
       base::MakeFixedFlatMap<ShapeContextTokens, ShapeSysTokens>({
           {ShapeContextTokens::kBadgeRadius, ShapeSysTokens::kXSmall},
-          {ShapeContextTokens::kButtonRadius, ShapeSysTokens::kFull},
+          {ShapeContextTokens::kButtonRadius, ShapeSysTokens::kMSmall},
           {ShapeContextTokens::kComboboxRadius, ShapeSysTokens::kSmall},
-          {ShapeContextTokens::kDialogRadius, ShapeSysTokens::kMediumSmall},
+          {ShapeContextTokens::kDialogRadius, ShapeSysTokens::kSmall},
           {ShapeContextTokens::kExtensionsMenuButtonRadius,
            ShapeSysTokens::kXSmall},
           {ShapeContextTokens::kFindBarViewRadius, ShapeSysTokens::kSmall},
-          {ShapeContextTokens::kMenuRadius, ShapeSysTokens::kMediumSmall},
-          {ShapeContextTokens::kMenuAuxRadius, ShapeSysTokens::kMediumSmall},
-          {ShapeContextTokens::kMenuTouchRadius, ShapeSysTokens::kMediumSmall},
+          {ShapeContextTokens::kMenuRadius, ShapeSysTokens::kSmall},
+          {ShapeContextTokens::kMenuAuxRadius, ShapeSysTokens::kSmall},
+          {ShapeContextTokens::kMenuTouchRadius, ShapeSysTokens::kSmall},
           {ShapeContextTokens::kOmniboxExpandedRadius, ShapeSysTokens::kMedium},
           {ShapeContextTokens::kTextfieldRadius, ShapeSysTokens::kSmall},
           {ShapeContextTokens::kSidePanelContentRadius,
-           ShapeSysTokens::kMedium},
+           ShapeSysTokens::kMediumSmall},
           {ShapeContextTokens::kSidePanelPageContentRadius,
            ShapeSysTokens::kSmall},
       });
@@ -229,6 +229,8 @@ int LayoutProvider::GetCornerRadiusMetri
   switch (token) {
     case ShapeSysTokens::kXSmall:
       return 4;
+    case ShapeSysTokens::kMSmall:
+      return 7;
     case ShapeSysTokens::kSmall:
       return 8;
     case ShapeSysTokens::kMediumSmall:
--- a/ui/views/layout/layout_provider.h
+++ b/ui/views/layout/layout_provider.h
@@ -183,6 +183,7 @@ enum class ShapeSysTokens {
   // mapping.
   kDefault,
   kXSmall,
+  kMSmall,
   kSmall,
   kMediumSmall,
   kMedium,
