--- a/ui/views/controls/button/md_text_button.cc
+++ b/ui/views/controls/button/md_text_button.cc
@@ -265,9 +265,8 @@ gfx::Insets MdTextButton::CalculateDefau
 
   int label_height = label()->GetPreferredSize({}).height();
   DCHECK_GE(target_height, label_height);
-  int top_padding = (target_height - label_height) / 2;
-  int bottom_padding = (target_height - label_height + 1) / 2;
-  DCHECK_EQ(target_height, label_height + top_padding + bottom_padding);
+  int top_padding = ((target_height - label_height) / 2) - 1;
+  int bottom_padding = ((target_height - label_height + 1) / 2) - 1;
 
   // TODO(estade): can we get rid of the platform style border hoopla if
   // we apply the MD treatment to all buttons, even GTK buttons?
