--- a/chrome/browser/ui/views/profiles/profile_menu_view.cc
+++ b/chrome/browser/ui/views/profiles/profile_menu_view.cc
@@ -719,6 +719,7 @@ void ProfileMenuView::BuildIdentityWithC
 }
 
 void ProfileMenuView::BuildAutofillSettingsButton() {
+  return;
   CHECK(!browser()->profile()->IsGuestSession());
   AddFeatureButton(
       l10n_util::GetStringUTF16(IDS_PROFILE_MENU_AUTOFILL_SETTINGS_BUTTON),
@@ -772,6 +773,7 @@ void ProfileMenuView::MaybeBuildChromeAc
 }
 
 void ProfileMenuView::MaybeBuildManageGoogleAccountButton() {
+  return;
   Profile* profile = browser()->profile();
   CHECK(!profile->IsGuestSession());
   signin::IdentityManager* identity_manager =
